src/ui/widget.cpp

```diff
@@ -499,7 +499,9 @@ Widget* Widget::pick(const gfx::Point& pt,
       (bounds().contains(pt))) {
     picked = this;
 
-    for (Widget* child : m_children) {
+    auto it = m_children.rbegin();
+    for (; it != m_children.rend(); ++it) {
+      const Widget* child = *it;
```

Comment:
Alright, this change breaks the Aseprite menus in single window UI mode...they do nothing when clicking the options. I'm investigating a bit why is this happening...

Reply:
> Alright, this change breaks the Aseprite menus in single window UI mode...they do nothing when clicking the options. I'm investigating a bit why is this happening...

This happens because the manager (app::CustomizedGuiManager) has their children not ordered as the rendering order. In this case, the menu goes first and the MainWindow goes last in the children list. Hence, when iterating in reverse order the MainWindow is picked and the mouse down message is sent to it instead of the menu.

Reply:
So...maybe is not a good idea to change the order...because it involves more changes to fix this...on the other hand, maybe it would be nice to fix how the manager is storing its children...but not sure how much work it would take and which other issues could arise.